vim-engine/src/main/kotlin/com/maddyhome/idea/vim/vimscript/model/commands/SourceCommand.kt

```diff
@@ -48,13 +48,23 @@ data class SourceCommand(val range: Range, val modifier: CommandModifier, val ar
     return ExecutionResult.Success
   }
 
-  private fun expandUser(path: String): String {
-    if (path.startsWith("~")) {
+  private fun String.vimExpanded(): String {
+    var expanded = this
+    if (startsWith("~")) {
       val home = System.getProperty("user.home")
       if (home != null) {
-        return home + path.substring(1)
+        expanded = home + substring(1)
```

Comment:
Hi, TBH I find a floating `substrint(1)` extremely confusing. I didn't even realize what the receiver of this function.
Please write an explicit receiver like `this.substring`.

Reply:
Same for `startsWith("~")`